tests/test_automatic_fields.py

```diff
@@ -0,0 +1,82 @@
+            "chat_search_interval_days": 500,
+        }
+
+        if original:
+            return return_value
+
+        # Start Date test needs the new connections start date to be prior to the default
+        assert self.start_date < return_value["start_date"]
+
+        # Assign start date to be the default
+        return_value["start_date"] = self.start_date
+        return return_value
+
+    def test_run(self):
+        """
+        - Verify we can deselect all fields except when inclusion=automatic, which is handled by base.py methods
+        - Verify that only the automatic fields are sent to the target.
+        - Verify that all replicated records have unique primary key values.
+        """
+
+        expected_streams = self.expected_streams()
+
+        conn_id = connections.ensure_connection(self)
+        found_catalogs = self.run_and_verify_check_mode(conn_id)
+        catalog_entries = [catalog for catalog in found_catalogs if catalog.get("stream_name") in expected_streams]
+        self.perform_and_verify_table_and_field_selection(
+            conn_id, catalog_entries, expected_streams, select_all_fields=False
+        )
+
+        # run initial sync
+        record_count_by_stream = self.run_and_verify_sync(conn_id)
+        synced_records = runner.get_records_from_target_output()
+
+        for stream in expected_streams:
+            with self.subTest(stream=stream):
+
+                expected_keys = self.expected_automatic_fields().get(stream)
+
+                data = synced_records.get(stream, {})
+                record_messages_keys = [set(row["data"].keys()) for row in data["messages"]]
+
+                self.assertGreater(
+                    record_count_by_stream.get(stream, -1),
+                    0,
+                    msg="The number of records is not over the stream max limit",
+                )
+                if stream == "chats":
+                    # chats stream has two types of records "offline_msgs" and "chat" both of them have different replication keys
+                    # the key "end_timestamp" is not available for "offline_msgs"
+                    # hence we need to verify the record has both or atleaset one key
+                    expected_keys_offline_msg = self.expected_automatic_fields().get(stream) - {"end_timestamp"}
+                    for actual_keys in record_messages_keys:
+                        if actual_keys == expected_keys:
+                            pass
+                        elif actual_keys == expected_keys_offline_msg:
+                            pass
+                        else:
+                            self.fail(
+                                f"Record of type: chat does not have the following automatic fields {expected_keys_offline_msg-actual_keys}"
+                            )
+                else:
+                    for actual_keys in record_messages_keys:
+                        self.assertSetEqual(expected_keys, actual_keys)
```

Comment:
For `chats` stream we can't distinguish between chat msgs and offline msgs where only automatic fields are selected, so we might miss scenario where chat msg type doesn't sync automatic field `end_timestamp`. To handle this situation we can do first full sync with all fields selected (if possible only for chats stream) and compare those records with automatic field selection sync records.

Reply:
this is fixed